File: AppointmentSchedulerAPI/layers/BusinessLogicLayer/Model/Appointment.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types;


namespace AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model
{
    public class Appointment
    {
        public int? Id { get; set; }
        public Guid? Uuid { get; set; }
        private DateTime _startDate;
        public DateTime StartDate
        {
            get => _startDate;
            set => _startDate = DateTime.SpecifyKind(value, DateTimeKind.Utc);
        }

        private DateTime _endDate;
        public DateTime EndDate
        {
            get => _endDate;
            set => _endDate = DateTime.SpecifyKind(value, DateTimeKind.Utc);
        }
        public AppointmentStatusType? Status { get; set; }
        public double? TotalCost { get; set; }
        private DateTime? _createdAt;
        public DateTime? CreatedAt
        {
            get => _createdAt;
            set => _createdAt = value.HasValue
                ? DateTime.SpecifyKind(value.Value, DateTimeKind.Utc)
                : null;
        }
        public Client? Client { get; set; } = new Client();
        public List<ScheduledService>? ScheduledServices { get; set; }
    }
}

File: AppointmentSchedulerWeb/src/view-model/business-entities/appointment.ts
import { AppointmentScheduledService } from "./appointment-scheduled-service";
import { Client } from "./client";
import { AppointmentStatusType } from "./types/appointment-status.types";

export class Appointment {
  uuid: string;
  startDate: Date;
  endDate: Date;
  status?: AppointmentStatusType;
  totalCost?: number;
  createdAt: Date;
  scheduledServices?: AppointmentScheduledService[]
  client?: Client;

  constructor(uuid: string, startDate: Date, endDate: Date, createdAt: Date, status?: AppointmentStatusType, totalCost?: number, scheduledServices?: AppointmentScheduledService[], client?: Client) {
    this.uuid = uuid;
    this.startDate = startDate;
    this.endDate = endDate;
    this.status = status;
    this.totalCost = totalCost;
    this.createdAt = createdAt;
    //this.assistants = assistants;
    this.scheduledServices = scheduledServices;
    this.client = client;
  }
}


class OccupiedTimeRange {
  startTime: string;
  endTime: string;

  constructor(startTime: string, endTime: string) {
    this.startTime = startTime;
    this.endTime = endTime;
  }
}

class Assistant {
  name: string;
  uuid: string;
  occupiedTimeRange: OccupiedTimeRange;

  constructor(name: string, uuid: string, occupiedTimeRange: OccupiedTimeRange) {
    this.name = name;
    this.uuid = uuid;
    this.occupiedTimeRange = occupiedTimeRange;
  }
}
